streaming/src/runtime_context.h

```diff
@@ -30,9 +33,36 @@ class RuntimeContext {
   inline void MarkMockTest() { is_mock_test_ = true; }
   inline bool IsMockTest() { return is_mock_test_; }
 
+  void InitMetricsReporter();
+
+  ///  It's periodic reporter entry for all runtime modules.
+  ///  \param metric_name, metric name
+  ///  \param value, metric value
+  ///  \param tags, metric tag map
+  void ReportMetrics(const std::string &metric_name, double value,
+                     const std::unordered_map<std::string, std::string> &tags = {});
+  void EnableTimer(std::function<void()> report_timer_handler);
+  void ShutdownTimer();
+
+ private:
+  void RunTimer();
+
+ protected:
+  std::unique_ptr<StreamingPerf> perf_metrics_reporter_;
+  std::function<void()> report_timer_handler_;
+
+  boost::asio::io_service async_io_;
```

Comment:
a `boost::asio::work` should be added?

Reply:
boost::asio::work is not recommended after boost 1.67. In our case, we can make sure all of behaviors will be finished.

> a `boost::asio::work` should be added?